pkg/clusteragent/admission/mutate/agent_sidecar/agent_sidecar.go

```diff
@@ -182,6 +206,37 @@ func (w *Webhook) injectAgentSidecar(pod *corev1.Pod, _ string, _ dynamic.Interf
 	return podUpdated, nil
 }
 
+func (w *Webhook) getDefaultSidecarInitTemplate() *corev1.Container {
+	return &corev1.Container{
+		Image:           fmt.Sprintf("%s/%s:%s", w.containerRegistry, w.imageName, w.imageTag),
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Name:            "copy-datadog-config",
+		Command:         []string{"sh", "-c", "cp -R /etc/datadog-agent/* /agent-config/"},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "agent-config",
+				MountPath: "/agent-config",
+			},
+		},
+	}
+}
+
+func (w *Webhook) getDefaultSidecarVolumeTemplate() *corev1.Volume {
+	return &corev1.Volume{
+		Name: "agent-config",
+		VolumeSource: corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{},
+		},
+	}
+}
+
+func (w *Webhook) addDefaultSidecarSecurity(container *corev1.Container) {
+	if container.SecurityContext == nil {
+		container.SecurityContext = &corev1.SecurityContext{ReadOnlyRootFilesystem: pointer.Ptr(true)}
+	}
+	container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{Name: "agent-config", MountPath: "/etc/datadog-agent"})
```

Comment:
Same suggestion, we can pass the volume name as an argument to avoid hardcoding the name